### PR TITLE
Add AAQ v2 faq search endpoint

### PR DESCRIPTION
**Pull request by Hlamallama**

### CODE REVIEW

File: aaq/views.py [hunk 1]
```diff
@@ -116,3 +117,37 @@ def check_urgency(request, *args, **kwargs):
...
+@renderer_classes((JSONRenderer,))
+def search(request, *args, **kwargs):
+    serializer = SearchSerializer(data=request.data)
+    serializer.is_valid(raise_exception=True)
+    query_text = serializer.validated_data["query_text"]
+    generate_llm_response = serializer.validated_data["generate_llm_response"]
+    query_metadata = serializer.validated_data["query_metadata"]
+    url = urllib.parse.urljoin(settings.AAQ_V2_API_URL, "search")
+    payload = {
+        "query_text": f"{query_text}",
```
Comment (erikh360) [inline aaq/views.py]: These probably don't have to be f-strings

File: aaq/views.py [hunk 2]
```diff
@@ -116,3 +117,37 @@ def check_urgency(request, *args, **kwargs):
...
+        "query_metadata": f"{query_metadata}",
+    }
+    headers = {
+        "Authorization": settings.AAQ_V2_AUTH,
+        "Content-Type": "application/json",
+    }
+
+    response = requests.request("POST", url, json=payload, headers=headers)
+
+    json_msg = {
```
Comment (erikh360) [inline aaq/views.py:168]: This looks like its just recreating the dict exactly how we got it from AAQ and returning it?

Shouldn't we build a message here like AAQ v1?

File: aaq/views.py [hunk 3]
```diff
@@ -116,3 +117,37 @@ def check_urgency(request, *args, **kwargs):
...
+        "query_id": response.json()["query_id"],
+        "llm_response": response.json()["llm_response"],
+        "search_results": response.json()["search_results"],
+        "feedback_secret_key": response.json()["feedback_secret_key"],
+        "debug_info": response.json()["debug_info"],
+        "state": response.json()["state"],
+    }
+
+    return_data = json_msg
```
Comment (erikh360) [inline aaq/views.py]: This extra variable isn't really required

**Review by erikh360 — COMMENTED** [review]
Left a few comments.

The commits and PR could be better named, ie "Add AAQ v2 faq search endpoint"

File: aaq/views.py (hunk 2 above)
Comment (DevChima) [reply aaq/views.py:168]: @Hlamallama Have have you seen this feedback?

Comment (Hlamallama) [reply aaq/views.py:168]: @DevChima yes, i've implemeted the feedback from Eric

**Review by DevChima — APPROVED** [review]

**Pull request merged**